python/whylogs/core/dataframe_wrapper.py

```diff
@@ -0,0 +1,158 @@
+from typing import Any, Callable, Dict, List, Optional, Tuple, Union
+
+from whylogs.core.stubs import pd, pl
+
+
+class DataFrameWrapper:
+    def __init__(self, pandas: Optional[pd.DataFrame] = None, polars: Optional[pl.DataFrame] = None):
```

Comment:
If we want to support additional data frame flavors, I'd make `DataFrameWrapper` an interface and have `_dataframe_or_dict()` return the appropriate subclass to wrap the input data frame type.